src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.Accessor.cs

```diff
@@ -129,23 +182,21 @@ private bool WriteStream(TKey key, Stream stream, Checksum? checksum, Cancellati
                     // Determine the appropriate data-id to store this stream at.
                     if (TryGetDatabaseId(connection, key, out var dataId))
                     {
-                        var (checksumBytes, checksumLength, checksumPooled) = GetBytes(checksum, cancellationToken);
+                        checksum ??= Checksum.Null;
+                        Span<byte> checksumBytes = stackalloc byte[Checksum.HashSize];
+                        checksum.WriteTo(checksumBytes);
```

Comment:
good observation by sharwell.  no need to pool the bytes for checksums.  it's much simpler to just stackalloc them as they're so small.